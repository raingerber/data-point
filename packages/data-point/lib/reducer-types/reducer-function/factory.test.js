/* eslint-env jest */
'use strict'

const rf = require('./factory')

test('reducer/reducer-function#isFunction', () => {
  expect(rf.isFunction('$foo')).toBe(false)
  expect(rf.isFunction('foo()')).toBe(false)
  expect(rf.isFunction(() => true)).toBe(true)
})

describe('reducer/reducer-function#validateFunction', () => {
  expect(rf.validateFunction(() => true)).toBe(true)
  expect(rf.validateFunction(a => true)).toBe(true)
  expect(rf.validateFunction((a, b) => true)).toBe(true)
  expect(() =>
    rf.validateFunction((a, b, c) => true)
  ).toThrowErrorMatchingSnapshot()
})

describe('reducer/reducer-function#create', () => {
  test('function body', () => {
    const reducerFunction = () => (acc, done) => done(null, acc.value * 2)
    const reducer = rf.create(reducerFunction)
    expect(reducer.body).toEqual(reducerFunction)
  })
})
