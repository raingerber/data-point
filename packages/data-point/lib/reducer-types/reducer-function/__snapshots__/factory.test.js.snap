// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `"Reducer functions must have an arity of 2 at most"`;
